xonsh/built_ins.py

```diff
@@ -555,6 +567,20 @@ def run_subproc(cmds, captured=True):
                 except PermissionError:
                     e = 'xonsh: subprocess mode: permission denied: {0}'
                     raise XonshError(e.format(cmd[0]))
+        _stdin_file = None
+        if (captured == 'object' and
+                stdin is not None and
+                __xonsh_env__['XONSH_STORE_STDIN'] and
```

Comment:
With the current implementation of the environment, there won't be any difference (since `__getitem__` falls back to the default value anyway).